### PR TITLE
Feature: Create Basic Authorization

### PR DESCRIPTION
**Pull request by camitaylor**
# [Create Basic Authorization](https://trello.com/c/1eo5dM3d/24-create-basic-authorization)

## A short description of what this PR does.
As a User, I want to click login button, so I can view protected routes.
As a Developer, I want an Authorization system, so I can protect routes from users. 

## How was it tested?
This was linted with Yarn and tested with Jest.

Screen Shot 2021-08-23 at 12 15 37 PM

### Yarn Lint
Screen Shot 2021-08-17 at 10 43 48 AM

### Checklist
- [x] I acknowledge that all my contributions will be made under the project's license
- [x] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [x] I have read the [Contribution Guidelines](https://github.com/aaronshivers/snapit-hacktoberfest/blob/main/CONTRIBUTING.md) and my PR follows them
- [x] I have titled the PR appropriately
- [x] I have updated my branch with the main branch
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added the necessary documentation about the functionality in the appropriate .md file
- [x] I have added inline documentation to the code I modified

### CODE REVIEW

File: src/pages/Login/Login.test.jsx [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    const wrapper = setup();
+    const loginButtonDisplay = findByTestAttribute(wrapper, "login-button");
+    expect(loginButtonDisplay.length).toBe(1);
+  });
+
+  test("button is clicked", () => {
+    const wrapper = setup();
+    const button = findByTestAttribute(wrapper, "login-button");
+    button.simulate("click");
+    expect(button.length).toBe(1);
```
Comment (aaronshivers) [inline src/pages/Login/Login.test.jsx:46]: 🔧 I think we need to check that the button has been clicked, not that there is a button here.

File: src/pages/Login/AuthButton.jsx [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      Welcome!{" "}
+      <button
+        type="button"
+        onClick={() => {
+          signout(() => history.push("/"));
+        }}
+      >
+        Sign out
+      </button>
+    </p>
```
Comment (aaronshivers) [inline src/pages/Login/AuthButton.jsx:20]: ♻️ Could you replace this with a Material-UI button? Also, the signout button.
https://material-ui.com/components/buttons/#button

**Review by aaronshivers — CHANGES_REQUESTED** [review]
🙂 It looks good, just a couple of minor adjustments are needed.

**Review by dylhack — APPROVED** [review]
LGTM after @aaronshivers review.

File: src/pages/Login/AuthButton.jsx (hunk 2 above)
Comment (camitaylor) [reply src/pages/Login/AuthButton.jsx:20]: Buttons replaced with Material UI

File: package.json [hunk 3]
```diff
@@ -17,7 +17,7 @@
     "@vitejs/plugin-react-refresh": "^1.3.5",
     "@wojtekmaj/enzyme-adapter-react-17": "^0.6.2",
     "clsx": "^1.1.1",
-    "enzyme": "^3.11.0",
```
Comment (aaronshivers) [inline package.json:20]: 🔧 We'd better leave enzyme in the project.

File: src/pages/Login/AuthButton.jsx (hunk 2 above)
Comment (aaronshivers) [reply src/pages/Login/AuthButton.jsx:20]: 🔧 Please link the commit. Also, I don't see the change.

File: src/pages/Login/Login.jsx [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+  return (
+    <div data-test="login-display">
+      <button
+        data-test="login-button"
+        type="button"
+        onClick={onLoginButtonClick}
+      >
+        Login
+      </button>
```
Comment (aaronshivers) [inline src/pages/Login/Login.jsx:24]: 🔧 make this a material button too

**Review by aaronshivers — CHANGES_REQUESTED** [review]

File: src/pages/Login/Login.test.jsx (hunk 1 above)
Comment (camitaylor) [reply src/pages/Login/Login.test.jsx:46]: Testing button click completed

File: package.json (hunk 3 above)
Comment (camitaylor) [reply package.json:20]: Enzyme added back in.

File: src/pages/Login/Login.jsx (hunk 4 above)
Comment (camitaylor) [reply src/pages/Login/Login.jsx:24]: Login and signout buttons are both using Material UI

File: src/pages/Login/AuthButton.jsx (hunk 2 above)
Comment (camitaylor) [reply src/pages/Login/AuthButton.jsx]: commit:  f45b314c2394685c40b9cac7c99d82ea5852972c

File: src/pages/Login/AuthButton.jsx [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+  return user.isAuthenticated ? (
+    <p>
+      Welcome!{" "}
+      <ButtonBase className={classes.image}>
+        <Button
+          href="#"
+          size="small"
+          variant="contained"
+          color="secondary"
+          className={classes.button}
```
Comment (aaronshivers) [inline src/pages/Login/AuthButton.jsx:30]: 🔧 Do we need `href` or `className` here?

File: src/pages/Login/Login.test.jsx [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+  });
+
+  test("should call mock function when button is clicked", () => {
+    const mockFn = jest.fn();
+    const button = shallow(
+      <Button data-test="login-button" onClick={mockFn} />
+    );
+    button.simulate("click");
+    expect(mockFn).toHaveBeenCalled();
+  });
```
Comment (aaronshivers) [inline src/pages/Login/Login.test.jsx:50]: 🔧 This looks like you're testing the button created on line 46, not the actual button from the login page.

**Review by aaronshivers — CHANGES_REQUESTED** [review]

File: src/pages/Login/Login.test.jsx (hunk 6 above)
Comment (camitaylor) [reply src/pages/Login/Login.test.jsx:50]: className is needed for the button to display.  href not needed so removed.  Also removed ButtonBase as it was throwing an error.  

Commit: d606eb88ec1b9224c41da3188eea5713125bc574.

File: src/pages/Login/AuthButton.jsx (hunk 5 above)
Comment (camitaylor) [reply src/pages/Login/AuthButton.jsx]: className is needed for the button to display.  href not needed so removed.  Also removed ButtonBase as it was throwing an error.  Commit: d606eb88ec1b9224c41da3188eea5713125bc574

File: src/pages/Login/Login.test.jsx (hunk 6 above)
Comment (camitaylor) [reply src/pages/Login/Login.test.jsx]: deleted failing test. Commit: a162079aafef755b3573e15045fc010449a1878e

File: src/pages/Login/Login.jsx [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+        Login with Github
+      </Button>
+      <AuthButton />
+    </div>
+  );
+};
+
+// Button.propTypes = {
+//   onClick: PropTypes.func.isRequired
+// }
```
Comment (aaronshivers) [inline src/pages/Login/Login.jsx:49]: 🔧 Can you update this commented code?

Comment (camitaylor) [reply src/pages/Login/Login.jsx]: Removed.  Commit: f48e77c5e5e80ecd6d2a211ab6d06fda5633d3ff

**Review by aaronshivers — APPROVED** [review]

**Pull request merged**